src/app/util/jwt-helper.ts

```diff
@@ -0,0 +1,45 @@
+import { HttpHeaders } from '@angular/common/http';
+
+export class JwtHelper {
+    private urlBase64Decode(str: string) {
+        let output = str.replace(/-/g, '+').replace(/_/g, '/');
+        switch (output.length % 4) {
+            case 0:
+                break;
+            case 2:
+                output += '==';
+                break;
+            case 3:
+                output += '=';
+                break;
+            default:
+                throw 'Illegal base64url string!';
+        }
+        return decodeURIComponent((<any>window).escape(window.atob(output)));
+    }
+
+    public decodeToken(token: string = '') {
+        if (token === null || token === '') { return { 'upn': '' }; }
+        const parts = token.split('.');
+        if (parts.length !== 3) {
+
+            throw new Error('JWT must have 3 parts');
+        }
+        const decoded = this.urlBase64Decode(parts[1]);
+        if (!decoded) {
```

Comment:
Codacy found an issue: [Expected indentation of 4 spaces but found 8.](https://app.codacy.com/manual/gaugsouza/petshow-frontend/pullRequest?prid=6332072)